.github/workflows/build.yaml

```diff
@@ -262,6 +265,45 @@ jobs:
           echo ${{ secrets.GITHUB_TOKEN }} | helm registry login -u ${{ github.actor }} --password-stdin ${{ env.OCI_REGISTRY }}
           helm push ${{ env.ARTIFACT_DIR }}/${{ env.HELM_PACKAGE_DIR }}/radius-${{ env.CHART_VERSION }}.tgz oci://${{ env.OCI_REGISTRY }}/${{ env.OCI_REPOSITORY }}
 
+  build-and-push-bicep-types:
+    name: Publish Radius bicep types to ACR
+    runs-on: ubuntu-latest
+        # Don't push on PR 
+    if: github.repository == 'radius-project/radius' && ((startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main'))
```

Comment:
The idea is that we only publish on merge to main or during a release in this workflow. The functional test workflow will also have a step to publish types to ACR and these will be tagged with the PR version. 

Open to suggestions if that doesn't sound like the right idea. Since the functional tests are where we'd consume the uploaded types (i.e. in all our test bicep templates), it made most sense to upload them as part of that workflow instead of here

Reply:
> Open to suggestions if that doesn't sound like the right idea. Since the functional tests are where we'd consume the uploaded types (i.e. in all our test bicep templates), it made most sense to upload them as part of that workflow instead of here

We could do that, or we could keep the bicep types on disk.

For the functional tests, we'll need to update/generate a bicep config to either:

- Reference the specific version associated with the PR from the OCI registry (like we do with magpie)
- Reference a file on disk 

For the purposes of this PR, I think you have the right idea 👍

Reply:
The plan was to generate a temporary bicep config in the functional test folder that has the PR-specific provider reference during pipeline runs. So we upload to ACR with the PR-specific tag and then the tests will pull from that provider tag. 

There will be a bicep config that permanently lives in the radius folder, but this will mostly be for development purposes and will always reference the latest tag by default